File: AI/Python/Assignment_01_Q1.py
parent = [('parent', 'karim', 'manjur'),
          ('parent', 'manjur', 'sohel'), ('parent', 'manjur', 'tofel'),
          ('parent', 'manjur', 'jerin'), ('parent', 'manjur', 'najnin'),
          ('parent', 'tofel', 'rifat'),  ('parent', 'jerin', 'raaj')]
          
X = str(input("Grandchildren:\n"))
print('Grandparent:', end='\n')
for i in range(len(parent)):
    if (parent[i][0] == 'parent') & (parent[i][2] == X):
        for j in range(len(parent)):
          if (parent[j][0] == 'parent') & (parent[i][1] == parent[j][2]):
            print(parent[j][1])
    i=i+1

File: AI/Prolog/Assignment_01_Q1.pl
parent('manjur','sohel'). parent('manjur','tufel').
parent('manjur','jerin'). parent('manjur','najnin').
parent('tufel','rifat'). parent('jerin','raaj').

grandparent(X, Z) :- parent(X, Y), parent(Y, Z).
findGp :- write('Grandchildren: '), read(X), write('Grandparent: '),
	grandparent(Gp, X), write(Gp), tab(5), fail.
findGp.
